From 1811528ca72a2b83924bc717c424f1301bf17c30 Mon Sep 17 00:00:00 2001
From: Andy Holmes <andrew.g.r.holmes@gmail.com>
Date: Sun, 1 Dec 2019 16:39:41 -0800
Subject: [PATCH] Notification Plugin: ignore the `silent` flag entirely

Upstream kdeconnect-android is tagging many notifications as `silent`,
causing them to never be shown. Since we already handle duplicates in
the Shell, we ignore that flag for now.

closes #720
---
 src/service/plugins/notification.js | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/service/plugins/notification.js b/src/service/plugins/notification.js
index b31a828ba..45ed2050c 100644
--- a/src/service/plugins/notification.js
+++ b/src/service/plugins/notification.js
@@ -135,16 +135,16 @@ var Plugin = GObject.registerClass({
 
     /**
      * Handle an incoming notification or closed report.
+     *
+     * FIXME: upstream kdeconnect-android is tagging many notifications as
+     *        `silent`, causing them to never be shown. Since we already handle
+     *        duplicates in the Shell, we ignore that flag for now.
      */
     _handleNotification(packet) {
         // A report that a remote notification has been dismissed
         if (packet.body.hasOwnProperty('isCancel')) {
             this.device.hideNotification(packet.body.id);
 
-        // A silent notification; silence it by aborting the icon transfer
-        } else if (packet.body.hasOwnProperty('silent') && packet.body.silent) {
-            this.device.rejectTransfer(packet);
-
         // A normal, remote notification
         } else {
             this.receiveNotification(packet);
